docs/2.8.0/docs/canton/usermanual/encrypted_private_key_storage.rst

```diff
@@ -0,0 +1,164 @@
+| `kms:CreateKey`                           | `cloudkms.cryptoKeyVersions.create`       |
++-------------------------------------------+-------------------------------------------+
+| `kms:TagResource`                         | `-`                                       |
++-------------------------------------------+-------------------------------------------+
+| `kms:Encrypt`                             | `cloudkms.cryptoKeyVersions.useToEncrypt` |
++-------------------------------------------+-------------------------------------------+
+| `kms:Decrypt`                             | `cloudkms.cryptoKeyVersions.useToDecrypt` |
++-------------------------------------------+-------------------------------------------+
+| `kms:DescribeKey`                         | `cloudkms.cryptoKeys.get`                 |
++-------------------------------------------+-------------------------------------------+
+
+If you plan to use cross-account key usage then the permissions for key rotation in Canton, namely
+`kms:CreateKey` or `cloudkms.cryptoKeyVersions.create`,
+do not have to be configured as they do not apply in that use case.
+
+Configure an Encrypted Private Key Storage in Canton
+----------------------------------------------------
+
+.. warning::
+    You cannot mix an encrypted private key storage configuration with an external private key storage
+    configuration.
+
+Encrypted private key storage support can be enabled for a new installation (i.e., during the node
+bootstrap) or for an existing deployment. **In both cases the keys are encrypted and stored in encrypted form
+in the Canton node's database in a transparent way without the need for the node manager to be actively engaged.**
+
+In the example below the encrypted private key storage
+integration is enabled for a participant node (called ``participant1``).
+The same applies for any other node, such as a sync domain manager, a mediator, or a sequencer.
+
+The most important setting that enables an encrypted private key storage using a
+KMS is ``type = kms``. This is shown below. If this is not specified, Canton
+stores the keys using its default approach, which is in unencrypted form.
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/resources/encrypted-store-enabled-tagged.conf
+   :language: none
+   :start-after: user-manual-entry-begin: PrivateKeyStoreConfig
+   :end-before: user-manual-entry-end: PrivateKeyStoreConfig
+
+There are two ways to choose the KMS wrapper key: (1) use an already existing KMS key or; (2)
+let Canton generate one.
+To use an already existing KMS key, you must specify its identifier. For example, for AWS KMS this can
+be one of the following:
+
+- Key id: `“1234abcd-12ab-34cd-56ef-1234567890ab”`
+- Key ARN (Amazon Resource Name): `“arn:aws:kms:us-east-1:1234abcd-12ab-34cd-56ef-1234567890ab”`
+- Key alias: `“alias/test-key”`
+
+Please be aware that an AWS KMS key needs to be configured with the following settings:
```

Comment:
IMO we'd benefit from splitting things more and not mix GCP / AWS on the same page. We think of it as just different KMS providers, but an actual user will never use both AWS and GCP at the same time, so they don't need to have information for both on the same page. It forces them to filter out the stuff they need from what they don't care about.

I'd suggest we split this further into 4 pieces:

- KMS Envelope Encryption
  - AWS Setup guide
  - GCP Setup guide
- Full KMS Integration
  - AWS Setup guide
  - GCP Setup guide

We can keep a fifth page that links to these pages and provides the general concept explanation of what envelope vs full KMS is, benefits etc... like you already have.

If I put myself in the step of a user, what I'd like to know is, in order:
- What type of KMS encryption does Canton offer? Pros and Const (Envelope vs Full)
- What KMS vendors are supported? (GCP, AWS)
- How do I set up the combination of the above that I have chosen for my deployment

I think the docs should reflect more or less this flow